docker/Dockerfile.build

```diff
@@ -1,56 +1,101 @@
+
+# Optimize builds. In particular, regenerate-stateful-test-disks.yml was reaching the
+# GitHub Actions time limit (6 hours), so we needed to make it faster.
+ENV RUSTFLAGS -O
+ENV CARGO_HOME /app/.cargo/
+# Build dependencies - this is the caching Docker layer!
+RUN cargo chef cook --release --features enable-sentry --recipe-path recipe.json
+
+ARG RUST_BACKTRACE=1
+ENV RUST_BACKTRACE ${RUST_BACKTRACE}
 
-ENV RUST_BACKTRACE full
-ENV CARGO_HOME /zebra/.cargo/
+COPY . .
+# Pre-download Zcash Sprout and Sapling parameters
+RUN cargo run --locked --release --features enable-sentry --bin zebrad download
+# Build zebra
+RUN cargo build --locked --release --features enable-sentry --bin zebrad
 
-RUN rustc -V; cargo -V; rustup -V
+FROM builder AS tester
+# Skip IPv6 tests by default, as some CI environment don't have IPv6 available
+ARG ZEBRA_SKIP_IPV6_TESTS=1
+ENV ZEBRA_SKIP_IPV6_TESTS ${ZEBRA_SKIP_IPV6_TESTS}
 
-COPY . .
+RUN cargo test --locked --release --features enable-sentry --workspace --no-run
 
-RUN cd zebrad/; cargo build --release --features enable-sentry
+CMD ["cargo" "test" "--locked" "--release" "--features" "enable-sentry" "--workspace"]
 
 # Runner image
-FROM debian:buster-slim AS zebrad-release
+FROM debian:bullseye-slim AS runtime
+COPY --from=builder /app/target/release/zebrad /usr/local/bin
+# TODO: do not hardcode the user /root/ even though is a safe assumption
+COPY --from=builder /root/.zcash-params /root/.zcash-params
 
 RUN apt-get update && \
-	apt-get install -y --no-install-recommends \
-	ca-certificates
-
-COPY --from=builder /zebra/target/release/zebrad /
+    apt-get install -y --no-install-recommends \
+    ca-certificates
 
 ARG CHECKPOINT_SYNC=true
 ARG NETWORK=Mainnet
 
-RUN printf "[consensus]\n" >> /zebrad.toml
-RUN printf "checkpoint_sync = ${CHECKPOINT_SYNC}\n" >> /zebrad.toml
-RUN printf "[metrics]\n" >> /zebrad.toml
-RUN printf "endpoint_addr = '0.0.0.0:9999'\n" >> /zebrad.toml
-RUN printf "[network]\n" >> /zebrad.toml
-RUN printf "network = '${NETWORK}'\n" >> /zebrad.toml
-RUN printf "[state]\n" >> /zebrad.toml
-RUN printf "cache_dir = '/zebrad-cache'\n" >> /zebrad.toml
-RUN printf "[tracing]\n" >> /zebrad.toml
-RUN printf "endpoint_addr = '0.0.0.0:3000'\n" >> /zebrad.toml
-RUN cat /zebrad.toml
-
-# Pre-download Zcash Sprout and Sapling parameters
-RUN /zebrad download
+RUN set -ex; \
+  { \
+    echo "[consensus]"; \
+    echo "checkpoint_sync = ${CHECKPOINT_SYNC}"; \
+    echo "[metrics]"; \
+    echo "endpoint_addr = '0.0.0.0:9999'"; \
+    echo "[network]"; \
+    echo "network = '${NETWORK}'"; \
+    echo "[state]"; \
+    echo "cache_dir = '/zebrad-cache'"; \
+    echo "[tracing]"; \
+    echo "endpoint_addr = '0.0.0.0:3000'"; \
+  } > "zebrad.toml"
 
 EXPOSE 3000 8233 18233
 
-ENV RUST_LOG debug
-ENV RUST_BACKTRACE full
-ENV SENTRY_DSN https://94059ee72a44420286310990b7c614b5@o485484.ingest.sentry.io/5540918
+ARG SHORT_SHA
+ENV SHORT_SHA $SHORT_SHA
+
+ARG SENTRY_DSN
+ENV SENTRY_DSN ${SENTRY_DSN}
```

Comment:
`SENTRY_DSN` is always injected, right?

Reply:
Yes, but it's complaining that the DSN is not correct. I'll be checking that when I'm working with the observability ticket.